In [2]:
library(haven)
wage2 <- read_dta("data_ps3/WAGE2.DTA")

In [3]:
dim(wage2)
colnames(wage2)

[1] 935  17

[1] "wage"    "hours"   "IQ"      "KWW"     "educ"    "exper"   "tenure" 
 [8] "age"     "married" "black"   "south"   "urban"   "sibs"    "brthord"
[15] "meduc"   "feduc"   "lwage"

In [6]:
#lecture's model (2)
reg1<-lm(lwage ~ educ + exper + tenure + married + south + urban + black + IQ, data=wage2)
summary(reg1)


Call:
lm(formula = lwage ~ educ + exper + tenure + married + south + 
    urban + black + IQ, data = wage2)

Residuals:
     Min       1Q   Median       3Q      Max 
-2.01203 -0.22244  0.01017  0.22951  1.27478 

Coefficients:
              Estimate Std. Error t value Pr(>|t|)    
(Intercept)  5.1764392  0.1280006  40.441  < 2e-16 ***
educ         0.0544106  0.0069285   7.853 1.12e-14 ***
exper        0.0141458  0.0031651   4.469 8.82e-06 ***
tenure       0.0113951  0.0024394   4.671 3.44e-06 ***
married      0.1997644  0.0388025   5.148 3.21e-07 ***
south       -0.0801695  0.0262529  -3.054 0.002325 ** 
urban        0.1819463  0.0267929   6.791 1.99e-11 ***
black       -0.1431253  0.0394925  -3.624 0.000306 ***
IQ           0.0035591  0.0009918   3.589 0.000350 ***
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 0.3632 on 926 degrees of freedom
Multiple R-squared:  0.2628,	Adjusted R-squared:  0.2564 
F-statistic: 41.27 on 8 and 926 DF,  p

In [ ]:
# use KWW instead of IQ (as a proxy for ability)
reg2<-lm(lwage ~ educ + exper + tenure + married + south + urban + black + KWW, data=wage2)
summary(reg2)


Call:
lm(formula = lwage ~ educ + exper + tenure + married + south + 
    urban + black + KWW, data = wage2)

Residuals:
     Min       1Q   Median       3Q      Max 
-2.04494 -0.21931 -0.00048  0.24163  1.26464 

Coefficients:
             Estimate Std. Error t value Pr(>|t|)    
(Intercept)  5.358797   0.113600  47.172  < 2e-16 ***
educ         0.057628   0.006838   8.428  < 2e-16 ***
exper        0.012228   0.003241   3.773 0.000172 ***
tenure       0.011072   0.002456   4.507 7.40e-06 ***
married      0.189461   0.039077   4.848 1.46e-06 ***
south       -0.091601   0.026156  -3.502 0.000484 ***
urban        0.175545   0.027032   6.494 1.36e-10 ***
black       -0.164267   0.038530  -4.263 2.22e-05 ***
KWW          0.005028   0.001819   2.764 0.005820 ** 
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 0.3642 on 926 degrees of freedom
Multiple R-squared:  0.2587,	Adjusted R-squared:  0.2523 
F-statistic: 40.39 on 8 and 926 DF,  p-value: <

note the similarity in R^2, i.e. how much of the variation in lwage is explained by the models, and in the ccoefficient in IQ (0.0035591) and KWW (0.005028).